### PR TITLE
Fix a bug in the locally-runnable proxy where headers were not copied

### PR DESCRIPTION
**Pull request by ojarjur**
This change fixes a bug in the locally-runnable version of the
inverting proxy where the headers from a proxied response were not
copied in to the response sent back to the frontend client.

That prevents things like set-cookie from working when testing local
changes.

That, in turn, was blocking the testing of a fix to
https://github.com/google/inverting-proxy/issues/31

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ import (
 	"time"
 
 	"context"
```
Comment (bsidhom) [inline server/server.go]: Might as well move context up with the standard go packages if we're reformatting this section.

File: server/server.go [hunk 2]
```diff
@@ -217,6 +218,9 @@ func (p *proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	case resp := <-pending.respChan:
 		defer resp.Body.Close()
+		for key, vals := range resp.Header {
+			w.Header()[key] = vals
```
Comment (bsidhom) [inline server/server.go]: Do we want to forward all headers or just cookies?

File: server/server.go (hunk 1 above)
Comment (bsidhom) [reply server/server.go]: Note that blindly sending all may include, e.g., `Content-Endoding` or other semantic headers.

File: server/server.go (hunk 2 above)
Comment (ojarjur) [reply server/server.go]: There's actually an RFC specifying which headers should be copied by proxies and which should be filtered (see https://www.w3.org/Protocols/rfc2616/rfc2616-sec13.html).

I was previously just being sloppy and copying everything, but I have now added the logic to distinguish between what should be copied and what should be filtered.

File: server/server.go (hunk 1 above)
Comment (ojarjur) [reply server/server.go]: Fixed the context import ordering.

Thanks for catching that.

File: server/server.go [hunk 3]
```diff
@@ -188,6 +188,20 @@ func (p *proxy) newID() string {
 	return fmt.Sprintf("%x", sum)
 }
 
+// isHopByHopHeader determines whether or not the given header name represents
+// a header that is specific to a single network hop and thus should not be
+// retransmitted by a proxy.
+//
+// See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#hbh
+func isHopByHopHeader(name string) bool {
+	switch n := strings.ToLower(name); n {
```
Comment (bsidhom) [inline server/server.go]: Why retain the unused variable here?

Comment (ojarjur) [reply server/server.go]: There are no unused variables. If there were, the code wouldn't build (the Go compiler treats unused variables as build errors).

Comment (bsidhom) [reply server/server.go]: `n` is only used in the sense that it's evaluated as a boolean. I mean to ask why not just use `strings.ToLower(name)` as the expression that's evaluated. Not a big deal either way.

Comment (bsidhom) [reply server/server.go]: s/evaluated as a boolean/evaluated for the switch statement/

Comment (bsidhom) [reply server/server.go]: s/evaluated as a boolean/evaluated for the switch statement/

Comment (ojarjur) [reply server/server.go]: Ah; I understand now.

I like that a lot better; thanks for the suggestion!

Done.

**Pull request merged**